message_parser/src/main/java/cumulus_message_adapter/message_parser/MessageParser.java

```diff
@@ -76,9 +79,15 @@ private String HandleMessage(String input, Context context, ITask task, SchemaLo
         }
         catch(Exception e)
         {
-            AdapterLogger.LogError(e.getMessage());
+            // Capture the stack trace as error output for logging.
+            StringWriter sw = new StringWriter();
+            PrintWriter pw = new PrintWriter(sw);
+            e.printStackTrace(pw);
+            String errorOutput = sw.toString();
+
+            AdapterLogger.LogError(errorOutput);
 
-            if(e.getClass().getSimpleName().contains("WorkflowError") || 
+            if(e.getClass().getSimpleName().contains("WorkflowError") ||
```

Comment:
Why are `WorkflowError` and `WorkflowException` being captured by string comparison? Can't they be captured in a `catch` block, like the following?
```java
try {
    ...
} catch (WorkflowError | WorkflowException e) {
    JsonElement jelement = ...
    ...
} catch (Exception e) {
    StringWriter sw = ...
    ...
}
```